lib/transaction.js

```diff
@@ -264,7 +264,7 @@ function makeTransactor(trx, connection, trxClient) {
   transactor.isTransaction = true;
   transactor.userParams = trx.userParams || {};
 
-  transactor.transaction = function(container, options) {
+  transactor.context.transaction = function(container, options) {
```

Comment:
It looks like this line was only working because of the way the proxy shortcuts had been setup.  In other words: one bug was canceling another bug.

So, now that the proxies are being setup correctly, we need to make sure we override the method in the correct place.  (on the `context` object rather than the `knex` function)